### ISSUE TITLE
throw null error

### ISSUE THREAD

**Issue by GuoSirius** [typescript-eslint/typescript-eslint#3132]
There are no errors in many parts of the project, which could run normally before, but now some files have this problem, not all of them

![image](https://user-images.githubusercontent.com/48358523/109617898-e943fd00-7b71-11eb-82f4-94121bcfe47a.png)

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
// Error
async function demo() {
if (options.length === 0) {
  try {
    options = (await this.getOptionsLists(queryString)) || []
  } catch { // cannot read property 'type' of null
    options = []
  }
}
}

// OK 
async function demo() {
if (options.length === 0) {
  try {
    options = (await this.getOptionsLists(queryString)) || []
  } catch (error) { // is OK
    options = []
  }
}
}

// OK 
async function demo() {
if (options.length === 0) {
  try {
    options = this.getOptionsLists(queryString) || [] // is OK
  } catch { 
    options = []
  }
}
}
```

**Expected Result**

**Actual Result**

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.15.2` |
| `TypeScript`                | `4.2.2` |
| `ESLint`                    | `7.21.0` |
| `node`                      | `14.16.0` |

**Comment by GuoSirius** [comment]
![image](https://user-images.githubusercontent.com/48358523/109617847-d7faf080-7b71-11eb-9e38-fd5dd24f1cf2.png)

**Comment by bradzacher** [comment]
As mentioned in your previous issue https://github.com/typescript-eslint/typescript-eslint/issues/3111

***This is not a bug in our project.*** 

The stack trace contains no references to packages from our project. 
The error stack trace is a babel stack.

Please don't raise another duplicate issue.
